portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/delete-comments")
+public class DeleteCommentServlet extends HttpServlet {
+
+  /** Deletes all comments from the page */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // create and prepare a query
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // deletes each comment
+    for (Entity entity : results.asIterable()) {
+        datastore.delete(entity.getKey());
+    }
+
+    response.sendRedirect("/index.html");
+  }
+
+}
```

Comment:
if I understand things correctly, you should be able to add this class to your existing servelet and it will allow you to use this end point as well as any other endpoint in your java file. I would try copying and pasting this class to step/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java and reloading that servelet and checking to see if you can hit that endpoint. If you can, then you wont need to stand up this servelet as well.

Reply:
on second thought, i might be wrong about this... I know java loves its classes and having files named after specific classes and having nothing else in there, so maybe actually the way you're doing it is best. in which case, ignore me lol.

Reply:
The tutorial asked us to add a new DataServlet for the url /delete-data, so I made a new file. but I could definitely try adding to DataServlet.java instead. Should I do that or stick with the tutorial? (if I'm even understanding what it's asking us to do)

Reply:
no I think you're doing it right and I'm wrong here. I was drawing on my experience with Python Flask for web based endpoints which is different than this so I think you're 100% correct here.

Reply:
awesome. I'm glad you questioned it though I'm following a lot of these instructions blindly